packages/node/src/indexer/apiPromise.connection.ts

```diff
@@ -123,6 +131,14 @@ export class ApiPromiseConnection
     await this.unsafeApi.disconnect();
   }
 
+  async updateRegisteredChainTypes(chainTypes: RegisteredTypes): Promise<void> {
+    this.unsafeApi = await ApiPromiseConnection.createApiPromise(
```

Comment:
This doesn't reuse the connection. If we have a websocket we don't want to create a new connection

Reply:
![image](https://github.com/user-attachments/assets/c670b506-0f06-4dc9-97e7-5a84d1e17521)

New approach : Extract current ws/http provider from current apiPromise instance, and reconstructed a new apiPromise